extensions/Worker.Extensions.Http.AspNetCore/src/HostBuilderExtensions.cs

```diff
@@ -25,15 +38,85 @@ public static IHostBuilder ConfigureAspNetCoreIntegration(this IHostBuilder buil
             builder.ConfigureWebHostDefaults(webBuilder =>
             {
                 webBuilder.UseUrls(HttpUriProvider.HttpUriString);
+
                 webBuilder.Configure(b =>
                 {
+                    b.UseRouting();
                     b.UseMiddleware<WorkerRequestServicesMiddleware>();
+
                     // TODO: provide a way for customers to configure their middleware pipeline here
+
+                    b.UseEndpoints(endpoints =>
+                    {
+                        string scriptRoot = Environment.GetEnvironmentVariable(FunctionsApplicationDirectoryKey) ??
+                           throw new InvalidOperationException("Cannot determine script root directory.");
+
+                        var metadataProvider = b.ApplicationServices.GetRequiredService<IFunctionMetadataProvider>();
+                        var metadata = metadataProvider.GetFunctionMetadataAsync(scriptRoot).GetAwaiter().GetResult();
+
+                        string routePrefix = GetRoutePrefixFromHostJson(scriptRoot) ?? DefaultRoutePrefix;
+
+                        foreach (var functionMetadata in metadata)
+                        {
+                            MapHttpFunction(endpoints, functionMetadata, routePrefix);
+                        }
+                    });
+
                     b.UseMiddleware<InvokeFunctionMiddleware>();
                 });
             });
 
             return builder;
         }
+
+        internal static void MapHttpFunction(IEndpointRouteBuilder endpoints, IFunctionMetadata functionMetadata, string routePrefix)
+        {
+            if (functionMetadata.RawBindings is null)
+            {
+                return;
+            }
+
+            var functionName = functionMetadata.Name ?? string.Empty;
+
+            foreach (var binding in functionMetadata.RawBindings)
+            {
+                var functionBinding = JsonSerializer.Deserialize<FunctionHttpBinding>(binding, _jsonSerializerOptions);
+
+                if (functionBinding is null)
+                {
+                    continue;
+                }
+
+                if (functionBinding.Type.Equals("httpTrigger", StringComparison.OrdinalIgnoreCase))
+                {
+                    string route = functionBinding.Route ?? functionName;
+
+                    endpoints.MapMethods($"{routePrefix}/{route}", functionBinding.Methods, InvokeFunctionMiddleware.InvokeInternal)
+                        .WithDisplayName(functionName);
+
+                    // no need to look at other bindings for this function
+                    break;
+                }
+            }
+        }
+
+        private static string? GetRoutePrefixFromHostJson(string scriptRoot)
+        {
+            string hostJsonPath = Path.Combine(scriptRoot, HostJsonFileName);
+
+            if (!File.Exists(hostJsonPath))
+            {
+                return null;
+            }
+
+            string hostJsonString = File.ReadAllText(hostJsonPath);
```

Comment:
Another future fixup will come here. We shouldn't have to read this file -- we'll update the host to send this along with process start and env reload.